src/schema/v2/me/myCollection.ts

```diff
@@ -137,12 +146,36 @@ export const MyCollection: GraphQLFieldConfig<any, ResolverContext> = {
         submissionIds,
         convectionGraphQLLoader
       )
+
+      let artistIDMediumTuples = artworks.map((artwork: any) => ({
+        artistId: artwork.artist._id,
+        medium: artwork.medium,
+      }))
+
+      // remove duplicates
+      artistIDMediumTuples = uniqWith(artistIDMediumTuples, isEqual)
+
+      const marketPriceInsights = await loadBatchPriceInsights(
+        artistIDMediumTuples,
+        vortexGraphqlLoaderWithVariables
+      )
+
       enrichArtworks(artworks, submissions)
 
-      return connectionFromArraySlice(artworks, options, {
+      const artworksWithInsights = artworks.map((artwork: any) => {
```

Comment:
What do you think about encapsulating all the logic into a method (e.g. `enrichArtworksWithPriceInsights` & renaming `enrichArtworks` to `enrichArtworksWithSubmissions`)?